src/library/trackcollectionmanager.cpp

```diff
@@ -624,3 +627,37 @@ bool TrackCollectionManager::updateTrackMood(
             mood);
 }
 #endif // __EXTRA_METADATA__
+
+// Replaces all genres for a given track by providing a list of names.
+bool TrackCollectionManager::replaceAllTrackGenres(
+        Track* pTrack, const QStringList& genreNames) const {
+    DEBUG_ASSERT_QOBJECT_THREAD_AFFINITY(this);
+    VERIFY_OR_DEBUG_ASSERT(pTrack) {
+        return false;
+    }
+
+    QList<DbId> genreIds;
+    genreIds.reserve(genreNames.size());
+
+    for (const QString& name : genreNames) {
+        const DbId genreId = m_pGenreDAO->getOrCreateGenre(name);
+        if (genreId.isValid()) {
+            genreIds.append(genreId);
+        }
+    }
+
+    // Update the track's associations in the database.
+    return m_pGenreDAO->setGenresForTrack(pTrack->getId(), genreIds);
+}
+
+// Retrieves all genres from the database.
+QList<Genre> TrackCollectionManager::getAllGenres() const {
+    DEBUG_ASSERT_QOBJECT_THREAD_AFFINITY(this);
+    return m_pGenreDAO->getAllGenres();
+}
+
+// Cleans unused genres from the database.
+int TrackCollectionManager::cleanupUnusedGenres() const {
```

Comment:
The function name is misleading because the genres itself are not cleaned. They are deleted. 
Do you plan to add more clean up tasks? Than it can become just "cleanUp()" else "deleteUnusedGenres()"